examples/platform/asset_groups.py

```diff
@@ -0,0 +1,194 @@
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+This example shows how to use the Asset Groups API and interact with related Policies and Devices.
+
+The SDK documentation is published on Read The Docs.  An Asset Groups Guide is available there.
+https://carbon-black-cloud-python-sdk.readthedocs.io
+
+"""
+
+import sys
+import time
+from cbc_sdk import CBCloudAPI
+from cbc_sdk.platform import AssetGroup, Policy, Device
+
+# To see the http requests being made, and the structure of the search requests enable debug logging
+# import logging
+# logging.basicConfig(level=logging.DEBUG)
+
+
+def preview_policy_rank_change(api):
+    """Demonstrate previewing the changes to effective policies on assets if policy ranking is changed.
+
+    Once Asset Groups have been created and policies assigned, the preview functions can be used to determine
+    the impact of changing asset groups or policy rankings.
+    This example finds the policy in the highest rank that has assets affected, and then moves it one position lower.
+    1 is the highest rank.
+    """
+    # Start by finding the highest ranked policy
+    print("\n\n Starting preview_policy_rank_change \n\n")
+    all_policies = list(api.select(Policy).all())
+    policy_highest_rank = None
+    policy_num_devices = 0
+
+    for p in all_policies:
+        tmp_policy_num_devices = len(api.select(Device).set_policy_ids([p.id]))
+        if tmp_policy_num_devices > 0:
+            if policy_highest_rank is None:
+                policy_highest_rank = p
+                policy_num_devices = tmp_policy_num_devices
+            elif p.position < policy_highest_rank.position:
+                policy_highest_rank = p
+                policy_num_devices = tmp_policy_num_devices
+
+    print("Policy {} with id = {}, is at rank {} and the policy affects {} members".
+          format(policy_highest_rank.name, policy_highest_rank.id, policy_highest_rank.position, policy_num_devices))
+
+    new_policy_position = policy_highest_rank.position + 1
+
+    # preview what would change if the policy at rank 1 was moved to position 2
+    changes = Policy.preview_policy_rank_changes(api, [(policy_highest_rank.id, new_policy_position)])
+    if len(changes) == 0:
+        print("No changes to effective policy would occur.")
+    else:
+        print("There are {} changes that would result from moving Policy {} from position {} to position {}."
+              .format(len(changes), policy_highest_rank.name, policy_highest_rank.position, new_policy_position))
+    i = 1
+    for c in changes:
```

Comment:
Instead of creating a counter, the following can be used: `for i, c in enumerate(changes, 1):` or `for i, name in enumerate(l):` if it is ok to start i from 0.

Reply:
thanks.